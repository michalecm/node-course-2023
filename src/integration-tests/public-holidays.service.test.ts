import * as service from '../services/public-holidays.service'

describe('public holidays service', () => {
    it('should get list of public holidays', async () => {
        expect(await service.getListOfPublicHolidays(2023, 'DE')).toMatchSnapshot()
    })

    it('should check that today is a public holiday', async () => {
        expect(await service.checkIfTodayIsPublicHoliday('DE')).toMatchSnapshot()
    })

    it('should fail to check that today is a public holiday', async () => {
         expect(await service.checkIfTodayIsPublicHoliday('DE')).toBe(false)
    })

    it('should get next public holiday', async () => {
        expect(await service.getNextPublicHolidays('DE')).toMatchSnapshot()
    })
})
