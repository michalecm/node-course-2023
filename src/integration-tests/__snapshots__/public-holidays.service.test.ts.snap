// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public holidays service should check that today is a public holiday 1`] = `false`;

exports[`public holidays service should get list of public holidays 1`] = `
[
  {
    "date": "2023-01-01",
    "localName": "Neujahr",
    "name": "New Year's Day",
  },
  {
    "date": "2023-01-06",
    "localName": "Heilige Drei Könige",
    "name": "Epiphany",
  },
  {
    "date": "2023-03-08",
    "localName": "Internationaler Frauentag",
    "name": "International Women's Day",
  },
  {
    "date": "2023-04-07",
    "localName": "Karfreitag",
    "name": "Good Friday",
  },
  {
    "date": "2023-04-09",
    "localName": "Ostersonntag",
    "name": "Easter Sunday",
  },
  {
    "date": "2023-04-10",
    "localName": "Ostermontag",
    "name": "Easter Monday",
  },
  {
    "date": "2023-05-01",
    "localName": "Tag der Arbeit",
    "name": "Labour Day",
  },
  {
    "date": "2023-05-18",
    "localName": "Christi Himmelfahrt",
    "name": "Ascension Day",
  },
  {
    "date": "2023-05-28",
    "localName": "Pfingstsonntag",
    "name": "Pentecost",
  },
  {
    "date": "2023-05-29",
    "localName": "Pfingstmontag",
    "name": "Whit Monday",
  },
  {
    "date": "2023-06-08",
    "localName": "Fronleichnam",
    "name": "Corpus Christi",
  },
  {
    "date": "2023-08-15",
    "localName": "Mariä Himmelfahrt",
    "name": "Assumption Day",
  },
  {
    "date": "2023-09-20",
    "localName": "Weltkindertag",
    "name": "World Children's Day",
  },
  {
    "date": "2023-10-03",
    "localName": "Tag der Deutschen Einheit",
    "name": "German Unity Day",
  },
  {
    "date": "2023-10-31",
    "localName": "Reformationstag",
    "name": "Reformation Day",
  },
  {
    "date": "2023-11-01",
    "localName": "Allerheiligen",
    "name": "All Saints' Day",
  },
  {
    "date": "2023-11-22",
    "localName": "Buß- und Bettag",
    "name": "Repentance and Prayer Day",
  },
  {
    "date": "2023-12-25",
    "localName": "Erster Weihnachtstag",
    "name": "Christmas Day",
  },
  {
    "date": "2023-12-26",
    "localName": "Zweiter Weihnachtstag",
    "name": "St. Stephen's Day",
  },
]
`;

exports[`public holidays service should get next public holiday 1`] = `
[
  {
    "date": "2023-10-03",
    "localName": "Tag der Deutschen Einheit",
    "name": "German Unity Day",
  },
  {
    "date": "2023-10-31",
    "localName": "Reformationstag",
    "name": "Reformation Day",
  },
  {
    "date": "2023-11-01",
    "localName": "Allerheiligen",
    "name": "All Saints' Day",
  },
  {
    "date": "2023-11-22",
    "localName": "Buß- und Bettag",
    "name": "Repentance and Prayer Day",
  },
  {
    "date": "2023-12-25",
    "localName": "Erster Weihnachtstag",
    "name": "Christmas Day",
  },
  {
    "date": "2023-12-26",
    "localName": "Zweiter Weihnachtstag",
    "name": "St. Stephen's Day",
  },
  {
    "date": "2024-01-01",
    "localName": "Neujahr",
    "name": "New Year's Day",
  },
  {
    "date": "2024-01-06",
    "localName": "Heilige Drei Könige",
    "name": "Epiphany",
  },
  {
    "date": "2024-03-08",
    "localName": "Internationaler Frauentag",
    "name": "International Women's Day",
  },
  {
    "date": "2024-03-29",
    "localName": "Karfreitag",
    "name": "Good Friday",
  },
  {
    "date": "2024-03-31",
    "localName": "Ostersonntag",
    "name": "Easter Sunday",
  },
  {
    "date": "2024-04-01",
    "localName": "Ostermontag",
    "name": "Easter Monday",
  },
  {
    "date": "2024-05-01",
    "localName": "Tag der Arbeit",
    "name": "Labour Day",
  },
  {
    "date": "2024-05-09",
    "localName": "Christi Himmelfahrt",
    "name": "Ascension Day",
  },
  {
    "date": "2024-05-19",
    "localName": "Pfingstsonntag",
    "name": "Pentecost",
  },
  {
    "date": "2024-05-20",
    "localName": "Pfingstmontag",
    "name": "Whit Monday",
  },
  {
    "date": "2024-05-30",
    "localName": "Fronleichnam",
    "name": "Corpus Christi",
  },
  {
    "date": "2024-08-15",
    "localName": "Mariä Himmelfahrt",
    "name": "Assumption Day",
  },
  {
    "date": "2024-09-20",
    "localName": "Weltkindertag",
    "name": "World Children's Day",
  },
]
`;
